pkgs/applications/misc/ulauncher/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv, fetchurl, python27Packages,
+  gnome3, gobjectIntrospection, wrapGAppsHook, gtk3, webkitgtk,
+  libnotify, keybinder3, libappindicator, intltool, wmctrl }:
+
+python27Packages.buildPythonApplication rec  {
+  pname = "ulauncher";
+
+  versionNumber = "4.3.2";
+  version = "${versionNumber}.r8";
+
+  src = fetchurl {
+    url = "https://github.com/Ulauncher/Ulauncher/releases/download/${version}/ulauncher_${version}.tar.gz";
+    sha256 = "0494326s84nvyhblcghxzidp2qf0v0l61kyrf8a92qqryqnzmwip";
+  };
+
+  nativeBuildInputs = with python27Packages; [
+    dbus-python
+    distutils_extra
+    gobjectIntrospection
+    gnome3.defaultIconTheme
+    pygtk
+    intltool
+    keybinder3
+    libappindicator
+    mock
+    notify
+    libnotify
+    paramiko
+    pygobject3
+    pyinotify
+    pysqlite
+    python-Levenshtein
+    pyxdg
+    websocket_client
+    webkitgtk
+    wmctrl
+    wrapGAppsHook
+  ];
+
+  pythonPath = nativeBuildInputs;
+
+  propagatedBuildInputs = pythonPath;
+
+  doCheck = false;
+
+  patchPhase = ''
+    substituteInPlace setup.py --replace "target_data + 'share/ulauncher/'" "'$out/share/ulauncher/'"
+    substituteInPlace ulauncher/config.py --replace "VERSION" "${version}"
+  '';
+
+  postInstall = ''
```

Comment:
This is unnecessary or, in case of `$PATH`, even harmful. You can rely on `wrapGAppsHook`.

Reply:
I assume that you are referring to `wrapProgram`. Weird that it works without it now; I couldn't get it to work before.

I still cannot figure out a way to have `ulauncher-toggle` see `wmctrl` when I remove the following:
```
wrapProgram $out/bin/ulauncher-toggle \
  --prefix PATH : "$PATH"
```

Reply:
I prefer patching the sources:

https://github.com/NixOS/nixpkgs/blob/0f3146eda73d21a1d3dd6b4a4ed593cab4e1650b/pkgs/desktops/gnome-3/core/gnome-shell/default.nix#L58-L61

But you can also use

```
gappsWrapperArgs+=(--prefix PATH : "${stdenv.lib.makeBinPath [ wmctrl ]}")
```

That will be picked up by `wrapGAppsHook` and will not include everything on the `PATH` (compiler and other stuff).

Reply:
> But you can also use... gappsWrapperArgs

Fixed.

> I prefer patching the sources

How do I access `$out` for substitution in a git diff?

Reply:
I think with `substituteAll` it doesn't work.

So you have to apply the patch normally
and then do
```
substituteInPlace $file --subst-var out
```

Reply:
The latest commit should fix this https://github.com/NixOS/nixpkgs/pull/57859/commits/9d512365a96a62dcc3b40eaef31c7b66c1c14d78